### PR TITLE
ci: add GitHub action to ensure that the CMS branch is set properly

### PR DESCRIPTION
**Pull request by greatislander**
* [x] This pull request has been linted by running `npm run lint` without errors
* [x] This pull request has been tested by running `npm run start` and reviewing affected routes
* [x] This pull request has been built by running `npm run build` without errors
* [x] This isn't a duplicate of an existing pull request

## Description

This PR adds a GitHub Action which monitors the Netlify CMS configuration file for changes and automatically adjusts the configuration to target the `main` branch for production and to target the `dev` branch for all other environments.

## Steps to test

1. Commit a change to the `src/admin/config.yml` to either `main` or `dev`.
2. Observe the results of the GitHub action.

**Expected behavior:** The CMS branch configuration should always be reset to `main` on `main` and `dev` on `dev`.

## Additional information

Uses [yq](https://mikefarah.gitbook.io/yq/) for parsing and updating the Netlify CMS configuration file.

## Related issues

Related to 54d638faa3f7aaadf1beeab87dd491819d4d56ab.

### CODE REVIEW

**Comment by cindyli** [comment]
I tested with a remote `test` branch in my own repo. The string gets written into the config.yml is [`$TARGET_BRANCH`](https://github.com/cindyli/wecount.inclusivedesign.ca/blob/test/src/admin/config.yml#L3) rather than the expected [`fromTest`](https://github.com/cindyli/wecount.inclusivedesign.ca/blob/test/.github/workflows/ensure-cms-branch.yml#L22).

File: .github/workflows/ensure-cms-branch.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+name: Ensure correct CMS branch
+
+on:
+  push:
+    branches:
+      - main
+      - dev
+    paths:
+      - 'src/admin/config.yml' # Skip this action unless the Netlify CMS config file has changed.
```
Comment (cindyli) [inline .github/workflows/ensure-cms-branch.yml:9]: Can we use double quotes when possible for consistency? Thanks.

Comment (greatislander) [reply .github/workflows/ensure-cms-branch.yml]: These quotes aren't actually required, so I'll remove them.

File: src/admin/config.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 backend:
   name: git-gateway
-  branch: dev
```
Comment (cindyli) [inline src/admin/config.yml:3]: This test renaming can be removed.

File: .github/workflows/ensure-cms-branch.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  ensure-cms-branch:
+    runs-on: ubuntu-latest
+    env:
+      target_branch: "dev" # Default to dev branch
+    steps:
+    - uses: actions/checkout@v2
+    - name: Adjust target branch for environment
+      if: endsWith(github.ref, '/main') # If the commit is in
```
Comment (cindyli) [inline .github/workflows/ensure-cms-branch.yml:19]: single quotes -> double quotes.

File: .github/workflows/ensure-cms-branch.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+jobs:
+  ensure-cms-branch:
+    runs-on: ubuntu-latest
+    env:
+      target_branch: "dev" # Default to dev branch
+    steps:
+    - uses: actions/checkout@v2
+    - name: Adjust target branch for environment
+      if: endsWith(github.ref, "/main") # If the commit is in
```
Comment (cindyli) [inline .github/workflows/ensure-cms-branch.yml:19]: Sorry about my ignorance about the syntax of Github Actions. It seems [strings have to use single quotes](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#literals). Using double quotes [breaks the job](https://github.com/cindyli/wecount.inclusivedesign.ca/actions/runs/741876536).

Comment (greatislander) [reply .github/workflows/ensure-cms-branch.yml]: Oh, interesting—I wasn't aware of that!

**Review by cindyli — APPROVED** [review]

**Comment by cindyli** [comment]
A brilliant idea. Thanks, @greatislander!

**Pull request merged**